xbmc/settings/DisplaySettings.cpp

```diff
@@ -219,6 +219,7 @@ void CDisplaySettings::Clear()
   CSingleLock lock(m_critical);
   m_calibrations.clear();
   m_resolutions.clear();
+  m_resolutions.insert(m_resolutions.begin(), RES_CUSTOM, RESOLUTION_INFO());
```

Comment:
can be a push_back.

Reply:
Yep, thanks.

Reply:
I dont agree, this is filling all entries until RES_DESKTOP (exclusive) with an empty resolution

Reply:
Sure, this is the intention. Erasing all elements and filling the vector with RES_CUSTOM elements. This is, what the ctor of this class does and was forgotten here. If CDisplaySettings::Clear() gets called and after this CDisplaySettings::AddResolutionInfo() which is supposed to add custom resos which must inserted into the vector at index RES_CUSTOM + n would not work.